### PR TITLE
PMD 6.0.0

### PR DESCRIPTION
**Pull request by TWiStErRob**
I suggest a Squash Merge again. I suggest you take a look at the two revert commits and then the extra commits I added later. Please make sure you review thoroughly, hopefully rebase didn't mess anything up. Feel free to commit any amendments to the branch (you have write rights to all PR branches).

## Description
Upgrade PMD from 5.8.1 to 6.0.0 based on https://github.com/sjudd/glide/tree/pmd_6_0_0 branch rebased onto #2762. Mostly PMD triggered fixes and refactors.

## Motivation and Context
Better checks for finding hidden issues in the code.

### CODE REVIEW

File: library/src/main/java/com/bumptech/glide/load/resource/bitmap/RecyclableBufferedInputStream.java [hunk 1]
```diff
@@ -368,15 +368,15 @@ public synchronized void reset() throws IOException {
    */
   @Override
   public synchronized long skip(long byteCount) throws IOException {
```
Comment (TWiStErRob) [inline library/src/main/java/com/bumptech/glide/load/resource/bitmap/RecyclableBufferedInputStream.java]: swapped order to mimic java.io.InputStream#skip (i.e. don't throw when no bytes are skipped)

File: library/src/main/java/com/bumptech/glide/load/data/mediastore/ThumbnailStreamOpener.java [hunk 2]
```diff
@@ -66,37 +68,43 @@ public int getOrientation(Uri uri) {
   }
 
   public InputStream open(Uri uri) throws FileNotFoundException {
```
Comment (TWiStErRob) [inline library/src/main/java/com/bumptech/glide/load/data/mediastore/ThumbnailStreamOpener.java]: extracted cursor handling to separate method and move up validation to look like guards

File: library/src/main/java/com/bumptech/glide/load/model/LazyHeaders.java [hunk 3]
```diff
@@ -42,28 +43,32 @@
...
-          if (i != factories.size() - 1) {
-            sb.append(',');
-          }
-        }
-      }
-      String values = sb.toString();
+      String values = buildHeaderValue(entry.getValue());
       if (!TextUtils.isEmpty(values)) {
-        combinedHeaders.put(entry.getKey(), sb.toString());
+        combinedHeaders.put(entry.getKey(), values);
```
Comment (TWiStErRob) [inline library/src/main/java/com/bumptech/glide/load/model/LazyHeaders.java]: extracted method to simplify code, and remove duplicate `sb.toString` building.

File: library/src/main/java/com/bumptech/glide/load/resource/bitmap/DefaultImageHeaderParser.java [hunk 4]
```diff
@@ -274,19 +270,16 @@ private static int parseExifSegment(RandomAccessReader segmentData) {
 
     int firstIfdOffset = segmentData.getInt32(headerOffsetSize + 4) + headerOffsetSize;
     int tagCount = segmentData.getInt16(firstIfdOffset);
-
-    int tagOffset, tagType, formatCode, componentCount;
```
Comment (TWiStErRob) [inline library/src/main/java/com/bumptech/glide/load/resource/bitmap/DefaultImageHeaderParser.java]: didn't just split them up, but moved them close to usage, thus can be final

**Comment by sjudd** [comment]
LGTM, thanks for cleaning this up!

**Pull request merged**